mkt/reviewers/forms.py

```diff
@@ -204,6 +210,14 @@ def clean_status(self):
 
         return mkt.STATUS_CHOICES_API_LOOKUP.get(status, mkt.STATUS_PENDING)
 
+    def clean(self):
+        dev_and_device = self.cleaned_data.pop('dev_and_device', '').split('+')
+        self.cleaned_data['dev'] = dev_and_device[0]
+        if len(dev_and_device) > 1:
+            self.cleaned_data['device'] = dev_and_device[1]
```

Comment:
It looks like this is using the slug, but the search filter is expecting the ID?